routes/auth.routes.js

```diff
@@ -0,0 +1,194 @@
+  ); // [ 1, 4, 5, 8 ]
+  const token = tokenArr.join(''); // "1458"
+
+  transporter
+    .sendMail({
+      from: '"Willkommen bei HelloCook " <myawesome@project.com>',
+      to: email,
+      subject: 'Bitte bestätige deine Anmeldung',
+      text: `Guten Tag, vielen Dank für deine Anmeldung. Wir freuen uns, dass wir dich bei HelloCook begrüßen dürfen!
+    Um Ihren Zugang zu bestätigen, klicken Sie einfach hier: ${process.env.EMAIL_LINK}${token}`,
+      html: `Um Ihren Zugang zu bestätigen, klicken Sie einfach: <a href="${process.env.EMAIL_LINK}${token}">hier!</a>`,
+    })
+    .then(() => {
+      const salt = bcrypt.genSaltSync(bcryptSalt);
+      const hashPass = bcrypt.hashSync(req.body.password, salt);
+      let user = new User({
+        email: req.body.email,
+        username: req.body.username,
+        password: hashPass,
+        token: token,
+      });
+      user.save().then(() => {
+        req.logIn(user, () => {
+          res.redirect('/willkommen');
+        });
+      });
+    });
+});
+
+router.get('/verify-email-link/:token', (req, res) => {
+  if (req.user.token === req.params.token) {
+    req.user.verifiedEmail = true;
+    req.user.save().then(() => {
+      // more professional : res.redirect and set a flash message before
+      req.flash(
+        'message',
+        'Super! Du hast deine E-Mail erfolgreich verifiziert.'
+      );
+      res.redirect('/willkommen');
+    });
+  }
+});
+
+// willkommen-seite
+router.get('/willkommen', (req, res) => {
+  const messages = req.flash('message');
+  const verifiedEmail = req.user.verifiedEmail;
+
+  Recipe.find().then((data) => {
+    const filteredData = data.filter((recipe) => {
+      if (req.query.level && req.query.dishType && req.query.nutrition) {
+        return (
+          recipe.level === req.query.level &&
+          recipe.dishType === req.query.dishType &&
+          recipe.nutrition === req.query.nutrition
+        );
+      } else if (req.query.dishType && req.query.nutrition) {
+        return (
+          recipe.dishType === req.query.dishType &&
+          recipe.nutrition === req.query.nutrition
+        );
+      } else if (req.query.nutrition && req.query.level) {
+        return (
+          recipe.nutrition === req.query.nutrition &&
+          recipe.level === req.query.level
+        );
+      } else if (req.query.level && req.query.dishType) {
+        return (
+          recipe.level === req.query.level &&
+          recipe.dishType === req.query.dishType
+        );
+      } else if (req.query.level) {
+        return recipe.level === req.query.level;
+      } else if (req.query.nutrition) {
+        return recipe.nutrition === req.query.nutrition;
+      } else if (req.query.dishType) {
+        return recipe.dishType === req.query.dishType;
+      } else {
+        return recipe;
+      }
+    });
+
+    let recipeCounter = filteredData.length;
+
+    let uniqueNutrition = [];
+    data.forEach((recipe) => {
```

Comment:
for this you could use `_.unique` from `lodash` or the `new Set()` that javascript supports now